postgrid/__init__.py

```diff
@@ -470,6 +479,156 @@ def construct_event(cls, payload, secret):
+            for inner_key, inner_value in value.items():
+                flatten(arr, f'{key}[{inner_key}]', inner_value)
+        elif isinstance(value, list):
+            for inner_value in value:
+                flatten(arr, f'{key}[]', inner_value)
+        elif isinstance(value, bool):
+            arr.append((key, 'true' if value else 'false'))
+        elif value is not None:
+            arr.append((key, value))
+
+    if data:
+        for key, value in data.items():
+            flatten(data_to_pass, key, value)
+
+    if params:
+        for key, value in params.items():
+            flatten(params_to_pass, key, value)
+
+    res = requests.post(
+        base + endpoint,
+        json=json,
+        data=data_to_pass,
+        params=params_to_pass,
+        headers=headers,
+    )
+    value = res.json()
+
+    try:
+        res.raise_for_status()
+    except requests.HTTPError as e:
+        raise AVError(status_code=res.status_code, message=value['message'])
+    return _av_convert_json_value(value)
+
+
+class RetrieveableAVResource:
+    @classmethod
+    def retrieve(cls, locals_):
+        assert 'cls' in locals_
+
+        locals_except_cls = {
+            key: value for key, value in locals_.items() if key != 'cls'
+        }
+
+        return _av_get(cls.endpoint, **locals_except_cls)
+
+
+class PostableAVResource:
+    @classmethod
+    def post(cls, **kwargs):
+
+        data = kwargs.get('data', None)
+        params = kwargs.get('params', None)
+        json = kwargs.get('json', None)
+
+        return _av_post_request(
+            av_base, cls.endpoint, json=json, data=data, params=params
+        )
+
+
+def _av_get(endpoint, **kwargs):
+    return _request(av_base, endpoint, method='GET', **kwargs)
+
+
+def create_dict(**kwargs):
+    return locals()['kwargs']
+
+
+class Address(BaseResource, RetrieveableAVResource, PostableAVResource):
+    endpoint = 'addver'
+
+    @classmethod
+    def lookup_info(cls):
+        return super().retrieve(locals())
+
+    @classmethod
+    def verify(cls, address=None):
+        cls.endpoint = 'addver/verifications'
+        return super().post(data=locals())
+
+    @classmethod
+    def autocomplete_previews(
+        cls, partial_street=None, country_filter=None, prov_instead_of_pc=None
+    ):
+        cls.endpoint = 'addver/completions'
+        return super().retrieve(locals())
+
+    @classmethod
+    def autocomplete_address(
+        cls,
+        partial_street=None,
+        city_filter=None,
+        state_filter=None,
+        pc_filter=None,
+        country_filter=None,
+        index=0,
+    ):
+        cls.endpoint = 'addver/completions'
+        return super().post(
+            data=create_dict(
+                partial_street=partial_street,
+                city_filter=city_filter,
+                state_filter=state_filter,
+                pc_filter=pc_filter,
+                country_filter=country_filter,
+            ),
```

Comment:
You can use a dictionary literal here instead:

```
{
  'partial_street': partial_street,
  'city_filter': city_filter,
  ...
}
```

Reply:
Fixed.